pkgs/by-name/ne/nexus-mods-app/package.nix

```diff
@@ -0,0 +1,125 @@
+{ _7zz
+, buildDotnetModule
+, copyDesktopItems
+, desktop-file-utils
+, dotnetCorePackages
+, fetchFromGitHub
+, fontconfig
+, lib
+, libICE
+, libSM
+, libX11
+, nexus-mods-app
+, runCommand
+, enableUnfree ? false # Set to true to support RAR format mods
+}:
+let
+  pname = "nexus-mods-app";
+  version = "0.4.1";
+  src = fetchFromGitHub {
+    owner = "Nexus-Mods";
+    repo = "NexusMods.App";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    hash = "sha256-FzQphMhiC1g+6qmk/R1v4rq2ldy35NcaWm0RR1UlwLA=";
+  };
+  meta = with lib; {
+    description = "Game mod installer, creator and manager";
+    mainProgram = "NexusMods.App";
+    homepage = "https://github.com/Nexus-Mods/NexusMods.App";
+    changelog = "https://github.com/Nexus-Mods/NexusMods.App/releases/tag/${src.rev}";
+    license = with licenses; [ gpl3 ];
+    maintainers = with maintainers; [ l0b0 ];
+    platforms = platforms.linux;
+  };
+  _7zzWithOptionalUnfreeRarSupport = _7zz.override {
+    inherit enableUnfree;
+  };
+in
+buildDotnetModule {
+  inherit meta pname src version;
+
+  projectFile = "NexusMods.App.sln";
+
+  nativeBuildInputs = [
+    copyDesktopItems
+  ];
+
+  nugetDeps = ./deps.nix;
+
+  dotnet-sdk = dotnetCorePackages.sdk_8_0;
+  dotnet-runtime = dotnetCorePackages.runtime_8_0;
+
+  preConfigure = ''
+    substituteInPlace Directory.Build.props \
+      --replace '</PropertyGroup>' '<ErrorOnDuplicatePublishOutputFiles>false</ErrorOnDuplicatePublishOutputFiles></PropertyGroup>'
+  '';
+
+  postPatch = ''
+    ln --force --symbolic "${lib.getExe _7zzWithOptionalUnfreeRarSupport}" src/ArchiveManagement/NexusMods.FileExtractor/runtimes/linux-x64/native/7zz
+  '';
+
+  makeWrapperArgs = [
+    "--prefix PATH : ${lib.makeBinPath [desktop-file-utils]}"
+    "--set APPIMAGE $out/bin/${meta.mainProgram}" # Make associating with nxm links work on Linux
+  ];
+
+  runtimeDeps = [
+    fontconfig
+    libICE
+    libSM
+    libX11
+  ];
+
+  executables = [
+    nexus-mods-app.meta.mainProgram
+  ];
+
+  doCheck = true;
+
+  dotnetTestFlags = [
+    "--environment=USER=nobody"
+    (
+      lib.strings.concatStrings [
+        "--filter="
+        (
+          lib.strings.concatStrings
+            (
+              lib.strings.intersperse "&" [
+                "Category!=Disabled"
+                "FlakeyTest!=True"
+                "RequiresNetworking!=True"
+                "FullyQualifiedName!=NexusMods.UI.Tests.ImageCacheTests.Test_LoadAndCache_RemoteImage"
+                "FullyQualifiedName!=NexusMods.UI.Tests.ImageCacheTests.Test_LoadAndCache_ImageStoredFile"
+              ]
+              ++ lib.optionals (! enableUnfree) [
+                "FullyQualifiedName!=NexusMods.Games.FOMOD.Tests.FomodXmlInstallerTests.InstallsFilesSimple_UsingRar"
+              ]
+            )
+        )
+      ]
+    )
+  ];
```

Comment:
Why are you running the tests?

Reply:
We run tests as part of our CI pipeline when PRs are submitted and merged, so published released have passing tests. Running these tests again here is very redundant. There are only a handful of tests that might be useful to run on. We run our tests on the `ubuntu-latest` GitHub Runner, and unless we're testing something super specific on Linux that might be different between that runner and NixOS, it's not really worth it to run these tests again.

Reply:
If particular tests are not useful in the nix package (or fail erroneously) they can be added to `disabledTests`. However generally speaking they'll be enough differences between the nix build and upstream that's it's useful to run tests, even if it feels redundant and wasteful.

There's also the implied responsibility shift; once a project is packaged by a distro, it becomes that distro's responsibility to ensure everything works correctly. So the distro may want to run tests on its package instead of just relying on upstream's CI.

Reply:
Why would I not? I'm trying to make sure the app works on the platform the user is trying to install to.